homeassistant/components/deconz.py

```diff
@@ -0,0 +1,198 @@
+"""
+Support for deCONZ support.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/deconz/
+"""
+
+import asyncio
+import os
+import logging
+import voluptuous as vol
+
+from homeassistant.config import load_yaml_config_file
+from homeassistant.const import (
+    CONF_API_KEY, CONF_EVENT, CONF_HOST, CONF_ID, CONF_PASSWORD, CONF_PORT,
+    CONF_USERNAME, EVENT_HOMEASSISTANT_STOP)
+from homeassistant.core import callback, EventOrigin
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers import discovery
+from homeassistant.util import slugify
+from homeassistant.util.json import load_json, save_json
+
+REQUIREMENTS = ['pydeconz==20']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'deconz'
+
+DECONZ_DATA = 'deconz_data'
+CONFIG_FILE = 'deconz.conf'
+
+CONF_SWITCH_AS_EVENT = 'switch_as_event'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_HOST): cv.string,
+        vol.Optional(CONF_API_KEY): cv.string,
```

Comment:
Should one either enter an API_KEY or USERNAME/PASSWORD ? If so, mark it as exclusive.

Reply:
With the new configurator integration, neither API_KEY nor USERNAME/PASSWORD is required, if the user follows the instructions to unlock the gateway. If he still wants to, they should be exclusive, yes. USERNAME/PASSWORD will only be considered if there is no API_KEY given.

After implementing auto-discovery, even the HOST can be made optional.

Reply:
Username and password are only used to generate API key. I will look into this exclusive thing.